### PR TITLE
Add support for merge queue ruleset JSON unmarshaling 

### PR DESCRIPTION
**Pull request by Matthew-Reidy**
Fixes: #3098.

Added support for respository ruleset merge_queue in JSON unmarshalling code with accompanying tests

### CODE REVIEW

File: AUTHORS [hunk 1]
```diff
@@ -485,3 +485,4 @@ zhouhaibing089 <zhouhaibing089@gmail.com>
 六开箱 <lkxed@outlook.com>
 缘生 <i@ysicing.me>
 蒋航 <hang.jiang@daocloud.io>
+Matthew Reidy <matthewreidy5@gmail.com>
```
Comment (gmlewis) [inline AUTHORS:488]: Please keep these lines sorted alphabetically, as an automated tool (run every few months) is usually used to update this list.

File: github/repos_rules.go [hunk 2]
```diff
@@ -134,7 +134,7 @@ func (r *RepositoryRule) UnmarshalJSON(data []byte) error {
 	r.Type = RepositoryRule.Type
 
 	switch RepositoryRule.Type {
-	case "creation", "deletion", "required_linear_history", "required_signatures", "non_fast_forward":
+	case "creation", "deletion", "required_linear_history", "required_signatures", "non_fast_forward", "merge_queue":
```
Comment (gmlewis) [inline github/repos_rules.go:137]: I know these weren't sorted before, but would you mind sorting them now, please?

File: github/repos_rules.go [hunk 3]
```diff
@@ -201,6 +201,7 @@ func (r *RepositoryRule) UnmarshalJSON(data []byte) error {
 		rawParams := json.RawMessage(bytes)
 
 		r.Parameters = &rawParams
+
```
Comment (gmlewis) [inline github/repos_rules.go:204]: No need for this extra blank line, please remove.

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @Matthew-Reidy !
Just a couple minor tweaks, please, then we should be ready for a second LGTM+Approval from any other contributor to this repo before merging.

**Comment by Matthew-Reidy** [comment]
Sure thing. I'll get I'll fix those items shortly

**Review by gmlewis — APPROVED** [review]
Thank you, @Matthew-Reidy !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

File: github/repos_rules.go [hunk 4]
```diff
@@ -210,6 +210,13 @@ func (r *RepositoryRule) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
+// NewMergeQueueRule creates a rule to only allow users with bypass permission to create matching refs.
```
Comment (liaodaniel) [inline github/repos_rules.go:213]:
```suggestion
// NewMergeQueueRule creates a rule to only allow merges via a merge queue.
```

**Comment by liaodaniel** [comment]
Just a minor comment to update but looks good otherwise

**Comment by Matthew-Reidy** [comment]
changes to that comment have been made

**Review by liaodaniel — APPROVED** [review]
LGTM

**Pull request merged**